app/grants/router.py

```diff
@@ -88,59 +93,161 @@ def get_queryset(self):
+
+        contributions_queryset = Contribution.objects.prefetch_related(
+            'subscription__grant'
+        ).filter(
+            subscription__grant=grant,
+            created_on__lte=to_timestamp,
+            created_on__gt=from_timestamp
+        )
+        all_contributions = Paginator(contributions_queryset, results_limit)
+
+        contributions_queryset = all_contributions.page(page)
+        transactions = txn_serializer(contributions_queryset, many=True).data
+
+        clr_serializer = CLRPayoutsSerializer
+        clr_payouts = clr_serializer(CLRMatch.objects.filter(grant=grant), many=True).data
+
+        return Response({
+            'transactions': transactions,
+            'clr_payouts': clr_payouts,
+            'metadata' : {
+                'grant_name': grant.title,
+                'from_timestamp': from_timestamp,
+                'to_timestamp': to_timestamp,
+                'count': all_contributions.count,
+                'current_page': page,
+                'num_pages': all_contributions.num_pages,
+                'has_next': all_contributions.page(page).has_next()
+            }
+
+        })
+
+
+    @action(detail=False)
+    @ratelimit(key='ip', rate='5/s')
+    def contributions_sent_report(self, request):
+        """Generate report for grant contributions made by an address"""
+
+        donor_serializer = DonorSerializer
+        results_limit = 30
+
+        # Validate input pararms
+        page = self.request.query_params.get('page', '1')
+        if not page or not page.isdigit():
+            page = 1
+
+        address = self.request.query_params.get('address', None)
+        if not address:
+            return Response({
+                'error': 'address is a mandatory parameter'
+            })
+
+        now = datetime.now()
+        to_timestamp = self.request.query_params.get('to_timestamp')
+        from_timestamp = self.request.query_params.get('from_timestamp')
+
+        # Check timestamp
+        if not from_timestamp:
+            from_timestamp = now - timedelta(days=30)
+        else:
+            try:
+                from_timestamp = datetime.strptime(from_timestamp, format)
+            except ValueError:
+                return Response({
+                    'error': 'from_timestamp is not in format YYYY-MM-DD'
+                })
+
+        if not to_timestamp:
+            to_timestamp = now
         else:
-            to_timestamp = datetime.now()
-        grants_queryset = grants_queryset.filter(subscriptions__subscription_contribution__created_on__lte=to_timestamp)
-        grants_queryset = grants_queryset.distinct()
+            try:
+                to_timestamp = datetime.strptime(to_timestamp, format)
+            except ValueError:
+                return Response({
+                    'error': 'to_timestamp is is not in the format YYYY-MM-DD'
+                })
+
+        if (to_timestamp - from_timestamp).days > 32:
```

Comment:
Same as above. This wouldn't catch the value `== 32`. Should be `>= 32` or `> 31`.